product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/vfs/VFSClient.java

```diff
@@ -0,0 +1,222 @@
+
+            try {
+                if (ftpClient.retrieveFile(sourceFilePath, outputStream1)) {
+                    log.info("File has been uploaded to " + targetFilePath + " successfully.");
+                }
+            } catch (IOException ex) {
+                throw new IOException("IO Exception occurred while downloading file ", ex);
+            }
+        } catch (FileNotFoundException ex) {
+            throw new IOException("File Not Found Exception occurred ", ex);
+        }
+    }
+
+    /**
+     * Function to list the set of directories on an FTP server
+     *
+     * @param filePath file path to list directories
+     * @return list of directories
+     * @throws IOException if listing set of directories fails
+     */
+    public FTPFile[] listDirectories(String filePath) throws IOException {
+        FTPFile[] directoryList;
+        try {
+            directoryList = ftpClient.listDirectories(filePath);
+        } catch (IOException ex){
+            throw new IOException("IO Exception occurred while listing directories", ex);
+        }
+        return directoryList;
+    }
+
+    /**
+     * Function to list files of a directory on an FTP server
+     *
+     * @param filePath file path to list files
+     * @return list of files
+     * @throws IOException if listing set of files fails
+     */
+    public List<FTPFile> listFiles(String filePath) throws IOException {
+        List<FTPFile> files = new ArrayList<>();
+        try{
+            FTPFile[] filesAndDirectories = ftpClient.listFiles(filePath);
+
+            for (FTPFile file : filesAndDirectories) {
+                if (!file.isDirectory()) {
+                    files.add(file);
+                }
+            }
+        } catch (IOException ex){
+            throw new IOException("IO Exception occurred while listing files", ex);
+        }
+        return files;
+    }
+
+    /**
+     * Function to list directories & files from a directory on an FTP server
+     *
+     * @param filePath file path to list directories & files
+     * @return list of directories & files
+     * @throws IOException if listing set of directories & files fails
+     */
+    public FTPFile[] listFilesAndDirectories(String filePath) throws IOException {
+        FTPFile[] fileDirectoryList;
+        try {
+            fileDirectoryList = ftpClient.listFiles(filePath);
+        } catch (IOException ex) {
+            throw new IOException("IO Exception occurred while listing files and directories", ex);
+        }
+        return fileDirectoryList;
+    }
+
+    /**
+     * Function to delete a file from an FTP server
+     *
+     * @param filePath  file path to delete
+     * @throws IOException if deleting the file fails
+     */
+    public void ftpDeleteFile(String filePath)
+            throws IOException {
+        try {
+            boolean deleted = ftpClient.deleteFile(filePath);
```

Comment:
Would it be a better idea to return the boolean indicating the status?

Reply:
fixed